mathphys/functions.py

```diff
@@ -197,3 +286,60 @@ def get_package_string(package):
     else:
         repo_str += f'{ver:s}'
     return repo_str
+
+
+# ------------------------- HELPER METHODS ------------------------------------
+_TYPES2SAVE = (int, float, complex, str, bytes, bool)
+_STRTYPES = {typ.__name__ for typ in _TYPES2SAVE}
+_NPTYPES = (_np.int_, _np.float_, _np.complex_, _np.bool_)
+
+
+def _save_recursive_hdf5(fil, path, obj, compress):
+    typ = type(obj)
+    if isinstance(obj, _np.ndarray):
+        fil.create_dataset(path, data=obj, compression=compress)
+    elif isinstance(obj, str):  # Handle SiriusPVName
+        fil[path] = str(obj)
+    elif isinstance(obj, _TYPES2SAVE + _NPTYPES):
+        fil[path] = obj
+    elif obj is None:
+        fil[path] = 'None'
+    elif typ in {list, tuple, set}:
+        path = path.strip('/') + '/'
+        for i, el in enumerate(obj):
+            _save_recursive_hdf5(fil, path + f'{i:d}', el, compress)
+    elif typ == dict:
+        path = path.strip('/') + '/'
+        for key, item in obj.items():
+            _save_recursive_hdf5(fil, path + key, item, compress)
+    else:
+        raise TypeError('Data type not valid: '+typ.__name__+'.')
+    fil[path].attrs['type'] = typ.__name__
+
+
+def _load_recursive_hdf5(fil):
+    """."""
```

Comment:
Users of the file who want to take advantage of something like partial loading could simply open the file themselves, right?
I think it makes sense to document in load_hdf5 that this means loading the whole file into memory (if I understood it correctly)

Reply:
Ok, I'll make this explicit in the comment, thanks!

Reply:
I'm already thinking of adding an option for the user to load only parts of the file, but I'll leave that for a future PR.